### PR TITLE
Integrate storage retry policy.

### PR DESCRIPTION
**Pull request by katmsft**

### CODE REVIEW

File: sdk/storage/azure-storage-files-datalake/src/datalake_path_client.cpp [hunk 1]
```diff
@@ -114,12 +118,15 @@ namespace Azure { namespace Storage { namespace Files { namespace DataLake {
     {
       policies.emplace_back(p->Clone());
     }
-    policies.emplace_back(
-        std::make_unique<Azure::Core::Http::RetryPolicy>(Azure::Core::Http::RetryOptions()));
+    StorageRetryWithSecondaryOptions dfsRetryOptions = options.RetryOptions;
+    dfsRetryOptions.SecondaryHostForRetryReads
+        = Details::GetDfsUriFromUri(options.RetryOptions.SecondaryHostForRetryReads);
```
Comment (Jinming-Hu) [inline sdk/storage/azure-storage-files-datalake/src/datalake_path_client.cpp:123]: `GetDfsUriFromUri` can return an empty string. Will everything still be able to work if so?

**Review by Jinming-Hu — APPROVED** [review]

File: sdk/storage/azure-storage-files-datalake/inc/azure/storage/files/datalake/datalake_utilities.hpp [hunk 2]
```diff
@@ -8,7 +8,7 @@
 
 namespace Azure { namespace Storage { namespace Files { namespace DataLake { namespace Details {
   std::string GetBlobUriFromUri(const std::string& uri);
-  std::string GetDfsUriFromUri(const std::string& uri);
+  std::string GetDfsUriFromUri(const std::string& uri, bool returnEmptyOnNotFound = true);
```
Comment (Jinming-Hu) [inline sdk/storage/azure-storage-files-datalake/inc/azure/storage/files/datalake/datalake_utilities.hpp:11]: This is really ugly.

Comment (katmsft) [reply sdk/storage/azure-storage-files-datalake/inc/azure/storage/files/datalake/datalake_utilities.hpp]: There is no better way to do so.

Comment (Jinming-Hu) [reply sdk/storage/azure-storage-files-datalake/inc/azure/storage/files/datalake/datalake_utilities.hpp:11]: Let me ask you one question. When is it useful to return an empty uri if the original uri doesn't contain `.dfs.`?

In [this](https://github.com/Azure/azure-sdk-for-cpp/blob/a898fd819aa0fa987358f751f246f4d8a8b85157/sdk/storage/azure-storage-files-datalake/src/datalake_file_system_client.cpp#L62) use case, `m_dfsUri` is empty if the original uri is a customized domain, how is an empty uri better than the original one?

Comment (katmsft) [reply sdk/storage/azure-storage-files-datalake/inc/azure/storage/files/datalake/datalake_utilities.hpp:11]: You are actually right. I'll remove this.

**Comment by katmsft** [comment]
Re-submitted https://github.com/Azure/azure-sdk-for-cpp/pull/722 to work-around the check enforcer hang.

**Pull request closed without merging**